v1/server/handlers/compress.go

```diff
@@ -61,17 +61,35 @@ type compressResponseWriter struct {
 	minlength     int
 }
 
-var gzipPool *sync.Pool
+var (
+	gzipPool                 *sync.Pool
+	gzipPoolMutex            sync.RWMutex
+	gzipPoolCompressionLevel int
+)
 
 func initGzipPool(compressionLevel int) {
-	if gzipPool == nil {
-		gzipPool = &sync.Pool{
-			New: func() any {
-				writer, _ := gzip.NewWriterLevel(io.Discard, compressionLevel)
-				return writer
-			},
-		}
+	gzipPoolMutex.RLock()
```

Comment:
Don't we also need to guard usage of `gzipPool` with the lock in other functions where it's used?

Reply:
Yeah, I only hit the issue here in tests but will check the other usages as they would be subject to the same issue.